src/Tokenizer/Analyzer/SwitchAnalyzer.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tokenizer\Analyzer;
+
+use PhpCsFixer\Tokenizer\Analyzer\Analysis\SwitchAnalysis;
+use PhpCsFixer\Tokenizer\Tokens;
+
+final class SwitchAnalyzer
+{
+    /** @var array<string, array<int>> */
+    private static array $cache = [];
+
+    public static function belongsToSwitch(Tokens $tokens, int $index): bool
+    {
+        if (!$tokens[$index]->equals(':')) {
+            return false;
+        }
```

Comment:
IMHO it is not intuitive - the intent of the method is checking if token belongs to `switch`, but in fact it supports only colons. Maybe we should name it differently or at least rename second param to `$colonIndex` to make it more explicit.

Reply:
It's definitely not intuitive, but even less intuitive would be to make it inconsistent with other analysers that do similar things - https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/blob/v3.39.0/src/Tokenizer/Analyzer/AlternativeSyntaxAnalyzer.php#L38 and https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/blob/v3.39.0/src/Tokenizer/Analyzer/GotoLabelAnalyzer.php#L26, wouldn't you agree?